cmd/exporter.go

```diff
@@ -86,41 +85,45 @@ the grid is greener or at locations where carbon intensity is lower.
 
 type Exporter struct {
 	apiClient      gridintensity.Provider
+	client         provider.Interface
 	provider       string
 	region         string
 	units          string
 	wattTimeClient watttime.Provider
 }
 
-func NewExporter(provider, region string) (*Exporter, error) {
+func NewExporter(providerName, regionName string) (*Exporter, error) {
 	var apiClient gridintensity.Provider
+	var client provider.Interface
 	var wattTimeClient watttime.Provider
 
 	var units string
 	var err error
 
-	if region == "" {
+	if regionName == "" {
 		return nil, fmt.Errorf("region must be set")
 	}
 
-	switch provider {
-	case carbonintensity.ProviderName:
-		apiClient, err = carbonintensity.New()
+	switch providerName {
+	case provider.CarbonIntensityOrgUK:
+		c := provider.CarbonIntensityUKConfig{}
+		client, err = provider.NewCarbonIntensityUK(c)
 		if err != nil {
 			return nil, err
 		}
-		units = "gCO2 per kWh"
-	case electricitymap.ProviderName:
+	case provider.ElectricityMap:
 		token := os.Getenv(electricityMapAPITokenEnvVar)
 		if token == "" {
-			return nil, fmt.Errorf("please set the env variable %q", electricityMapAPITokenEnvVar)
+			return nil, fmt.Errorf("%q env var must be set", electricityMapAPITokenEnvVar)
```

Comment:
@rossf7  - can we please make an issue to revisit all the error and command line copy?

I feel I can share my reckons on each PR, but there must be some good guidelines online somewhere we can refer to in future when reviewing PRs, so there's something that isn't as subjective as me giving rando comments  all the time.

Reply:
@mrchrisadams Yes, I changed this for consistency with other providers but my bad it makes the PR harder to review.

I've created https://github.com/thegreenwebfoundation/grid-intensity-go/issues/51 to revisit this. 

It also might make sense to use an external library like `juju/errors`
https://github.com/juju/errors